File: C Programming/problems on file handling/problem8.c
/*
8. Write a program which accepts file name from user and count number of white spaces from that file. 
	
	Input : Demo.txt 
	
	Output : Number of white spaces are 13 
	
	Author : Milind Nalawade

	Date : 1/09/2021

	Language : C programming
	
*/

#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>
#include<fcntl.h>

//Entry point function
int main()
{
	int fd=0,ret=0,cnt=0,i=0;
	char arr[100]={'\0'};
	
	printf("\n Enter the file name : ");
	scanf("%[^'\n']s",arr);

	fd=open(arr,O_RDONLY|O_CREAT,0777);
	if(fd==-1)
	{
		printf("\n file unalbe to open\n");
	close(fd);
	}

	while((ret=read(fd,arr,100))!=0)
	{
		while(arr[i]!='\0')
		{
			if(arr[i]==' ')
			{
				cnt++;
			}
			i++;
		}			
	}
	printf("\n Number of spaces in the file : %d \n",cnt);
	return 0;

}

File: Java Programming/problems on file handling/problem8.java
/*
8. Write a program which accepts file name from user and count number of white spaces from that file. 
	
	Input : Demo.txt 
	
	Output : Number of white spaces are 13 

	Author : Milind Nalawade

	Date : 01/09/2021

	Language : java 14.0.2

	
*/

import java.lang.*;
import java.util.*;
import java.io.*;

class problem8
{
	public static void main(String args[])
	{
		Scanner sobj =new Scanner(System.in);
		System.out.print("\n Enter file name : ");
		String str=sobj.nextLine();
		String line=null;
		int cnt=0;
		try 
		{
			FileReader fd=new FileReader(str);
			BufferedReader bf=new BufferedReader(fd);
			
			while((line = bf.readLine())!=null)
			{
				for(int i=0;i<=(line.length()-1);i++)
				{
					if(line.charAt(i)==' ')
					{
						cnt++;
					}
				}
			}
			bf.close();
		}
		catch(IOException e)
		{
			System.out.print("\n Error in reading the file !!! \n");
		}
		System.out.print("\n Number of spaces are : "+cnt+"\n");
	}
}
